### DISCUSSION TITLE
useA11y() not working as expected

### DISCUSSION THREAD

**Discussion by pi43r** [pmndrs/react-three-a11y#23] [Q&A] [answered]
Hi,

First of all: Really amazing library! 🦸‍♀️

I am using the [react-three-next starter](https://github.com/pmndrs/react-three-next) and am a bit confused about the setup.
If I see that correctly, the starter never uses `<A11yAnnouncer />` so I added it into `/components/layout/canvas.jsx`

Then I changed the hover function to use the booleans from this library, but it doesn't work.
See `components/canvas/Box.jsx` in this [sandbox](https://codesandbox.io/s/young-sun-f6o9j?file=/src/components/canvas/Box.jsx)

The expected behavior would be that the color changes to hotpink on hover. But there is no change.

**Comment by AlaricBaraou** [comment]
Hi !

Thanks I'm glad you appreciate this library :) 

I was taking a quick look and it started working.

Should I assume you found a solution to your problem ?

![hovertest](https://user-images.githubusercontent.com/7174039/122761110-299fc600-d29c-11eb-9fca-edf8c76d8bd6.gif)

**Comment by pi43r** [comment, marked as answer]
Ahhh, yes.
I think I answered my own question :)
For useA11y to work correct, it has to be encapsulated in it's own component like this `<A11y> <Component \> </A11y>`. 
Originally I was using it like this, which does not work.
```js
<A11y
    role='link'
    href={route}
    actionCall={() => {
      router.push(route)
    }}
  >
    <mesh
      ref={mesh}
      onPointerOver={(event) => setHover(true)}
      onPointerOut={(event) => setHover(false)}
    >
      <boxGeometry args={[1, 1, 1]} />
      <meshBasicMaterial
        // changed hovered to a11y functions
        //color={hovered ? 'hotpink' : color}
        color={a11y.focus || a11y.hover ? 'hotpink' : color}
      />
    </mesh>
</A11y>
```

**Reply by AlaricBaraou** [reply]
You're right, the starter doesn't do it this way to keep thing simple and minimal. 
Glad you figured it out !

**Comment by adamTheWordUnderscoreWillis** [comment]
Hey, 

I tried so hard to figure out what you meant by encapsulating it in its own component, but I could not get it to work. And example would be amazing.

**Reply by AlaricBaraou** [reply]
Hello @adamTheWordUnderscoreWillis 
I haven't updated / used the lib in years.
I need to find the time to update everything, I don't know how much it works with the latest releases.

Sorry I can't be more helpful at the moment.
